rust_src/src/callproc.rs

```diff
@@ -79,4 +83,53 @@ pub fn call_process_lisp(args: &mut [LispObject]) -> LispObject {
     })
 }
 
+#[lisp_fn(min = "3")]
+pub fn call_process_region(args: &mut [LispObject]) -> LispObject {
+    let mut start = args[0];
+    let mut end = args[1];
+    let mut infile = Qnil;
+    let spec = c_specpdl_index();
+
+    let empty_input = if start.is_string() {
+        LispStringRef::from(start).len_chars() == 0
+    } else if start.is_nil() {
+        let buffer = &ThreadState::current_buffer_unchecked();
```

Comment:
Why the reference?

Reply:
That was a mistake, I have a habit of forgetting the behaviour of `ExternalPtr`.